lib/bangladesh/to_patients.rb

```diff
@@ -56,35 +60,69 @@
   phone_number_authentication.save!
 end
 
+def report_failure(reason, params)
+  puts "#{ERR_LOG_TAG} ex_id: #{params[:external_id]} - Patient failed: #{reason}"
+end
+
+def report_success(params)
+  puts "ex_id: #{params[:external_id]} - Patient created"
+end
+
+def patient_gender(gender)
+  if ['m', 'M', 'male', 'Male', 'MALE'].include?(gender)
+    'male'
+  elsif ['f', 'F', 'female', 'Female', 'FEMALE'].include?(gender)
+    'female'
+  end
+end
+
 def national_id(value)
   # scientific notation
   if /e\+/.match?(value)
     base, exp = value.split("e+").map(&:to_i)
-    (base * exp**10).to_i.to_s
+    (base * exp ** 10).to_i.to_s
   else
     value
   end
 end
 
 def history(value)
-  case value.to_i
-  when 0
-    'unknown'
-  when 1
-    'yes'
-  end
+  return 'yes' if value.to_i == 1
+
+  'unknown'
 end
 
 def dosage(value)
   dosage_value = value.scan(/(\d|\.)/).join.to_f
   if dosage_value.to_i == dosage_value
-    "#{dosage_value.to_i} mg"     # 50.0 -> "50 mg"
+    "#{dosage_value.to_i} mg" # 50.0 -> "50 mg"
   else
-    "#{dosage_value} mg"          # 12.5 -> "12.5 mg"
+    "#{dosage_value} mg" # 12.5 -> "12.5 mg"
   end
 end
 
+def patient_recorded_at(blood_pressures)
+  date = blood_pressures.map { |bp| bp[:recorded_at] }.min
+  return DateTime.new(2019, 6, 1) if date.nil?
+  date
+end
+
 def create_patient(params)
+  if params[:gender].blank?
```

Comment:
Any idea why this doesn't work? Does the sheet have spaced strings?

Reply:
We moved the gender check to where the CSV is parsed to fix this